tests/test/generators.rs

```diff
@@ -0,0 +1,89 @@
+use super::*;
+
+#[test]
+fn generator_test() {
+    test! {
+        program {
+            #[auto] trait Send { }
+
+
+            struct StructOne {}
+            struct NotSend {}
+            struct SendSameLifetime<'a, 'b, T> { val: &'a T, other: &'b T }
+            impl<'a, T> Send for SendSameLifetime<'a, 'a, T> {}
+
+            struct SendAnyLifetime<'a, 'b, T> { val: &'a u8, other: &'b u8, field: T }
+
+            impl !Send for NotSend {}
+            struct StructThree<'a> { val: &'a () }
+
+            generator empty_gen<>[resume = (), yield = ()] {
+                upvars []
+                witnesses []
+            }
+
+            generator upvar_lifetime_restrict<T>[resume = (), yield = ()] {
+                upvars [T; StructOne]
+                witnesses for<'a, 'b> [SendSameLifetime<'a, 'b, T>]
+            }
+
+            generator send_any_lifetime<T>[resume = (), yield = ()] {
+                upvars []
+                witnesses for<'a, 'b> [SendAnyLifetime<'a, 'b, T>; u8]
```

Comment:
the `for` syntax here seems a bit misleading, I guess?

Reply:
Yeah, it's not great. Do you think `exists<>` would be better?